internal/k8s/converter.go

```diff
@@ -37,34 +37,14 @@ type DynamicClientHandler struct {
 }
 
 func (d *DynamicClientHandler) OnAdd(obj interface{}) {
-	obj, err := d.Converter.FromUnstructured(obj)
```

Comment:
Is it guaranteed that we'll never get Unstructured? If so, neat.

Reply:
Yeah we shouldn't since the controller-runtime is doing all of this for us now.